[ws-manager] nil deref in IsWorkspaceTimedOut

### Bug description

```
panic: runtime error: invalid memory address or nil pointer dereference [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x196cba6] goroutine 397 [running]: github.com/gitpod-io/gitpod/ws-manager/pkg/manager.(*Manager).isWorkspaceTimedOut(0xc0006e8580, 0xc001c4dc00, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...) github.com/gitpod-io/gitpod/ws-manager/pkg/manager/status.go:791 +0xf06 github.com/gitpod-io/gitpod/ws-manager/pkg/manager.(*Monitor).markTimedoutWorkspaces(0xc00049b5e0, 0x21197b0, 0xc0019784e0, 0x0, 0x0) github.com/gitpod-io/gitpod/ws-manager/pkg/manager/monitor.go:1023 +0x728 github.com/gitpod-io/gitpod/ws-manager/pkg/manager.(*Monitor).doHousekeeping(0xc00049b5e0, 0x2119740, 0xc0019784b0) github.com/gitpod-io/gitpod/ws-manager/pkg/manager/monitor.go:381 +0xc6 github.com/gitpod-io/gitpod/ws-manager/pkg/manager.(*Monitor).Start.func1(0xc00049b5e0) github.com/gitpod-io/gitpod/ws-manager/pkg/manager/monitor.go:122 +0x7c created by github.com/gitpod-io/gitpod/ws-manager/pkg/manager.(*Monitor).Start github.com/gitpod-io/gitpod/ws-manager/pkg/manager/monitor.go:120 +0x27c
```

### Steps to reproduce

run ws-manager

### Expected behavior

_No response_

### Example repository

_No response_

### Anything else?

_No response_

--- a/components/ws-manager/pkg/manager/status.go
+++ b/components/ws-manager/pkg/manager/status.go
@@ -787,7 +787,9 @@ func (m *Manager) isWorkspaceTimedOut(wso workspaceObjects) (reason string, err 
 		return decide(*lastActivity, m.Config.Timeouts.Interrupted, activityInterrupted)
 
 	case api.WorkspacePhase_STOPPING:
-		if status.Conditions.FinalBackupComplete != api.WorkspaceConditionBool_TRUE {
+		if isPodBeingDeleted(wso.Pod) && status.Conditions.FinalBackupComplete != api.WorkspaceConditionBool_TRUE {
+			// Beware: we apply the ContentFinalization timeout only to workspaces which are currently being deleted.
+			//         We basically don't expect a workspace to be in content finalization before it's been deleted.
 			return decide(wso.Pod.DeletionTimestamp.Time, m.Config.Timeouts.ContentFinalization, activityBackup)
 		} else if !isPodBeingDeleted(wso.Pod) {
 			// pods that have not been deleted have never timed out

